build.go

```diff
@@ -116,15 +76,9 @@ func Build(
 			"ruby_version": rubyVersion,
 		}
 
-		gemsLayer.SharedEnv.Default("BUNDLE_PATH", gemsLayer.Path)
-		gemsLayer.SharedEnv.Default("BUNDLE_WITHOUT", "development:test")
+		gemsLayer.SharedEnv.Default("BUNDLE_USER_CONFIG", filepath.Join(gemsLayer.Path, "config"))
```

Comment:
I see in the original issue #167 that you had recommended we set `BUNDLE_APP_CONFIG` to the layer path. Why are we opting for `BUNDLER_USER_CONFIG` instead? 

I see from the docs that `BUNDLE_APP_CONFIG` relates to the local config. Just wondering why we wouldn't go that route.

Reply:
After starting to implement this, I discovered that `BUNDLE_APP_CONFIG` is a recent addition to Bundler and isn't available before `v2.2`. Since we support older versions than that, we needed to choose a different mechanism. The `BUNDLE_USER_CONFIG` setting has been supported since at least `v1.17` and works similarly, so it seemed to good route to choose.

Reply:
Makes sense, thanks for the clarification